File: php/raindrops/raindrops.php
<?php

function raindrops($number)
{
    $numbers = range(1, $number);

    $factors = array_filter($numbers, function ($divider) use ($number) {
        if (($number % $divider) == 0) {
            return $divider;
        }
    });

    $intersect = array_intersect([ 3, 5, 7 ], $factors);

    if (empty($intersect)) {
        return "{$number}";
    }

    $result = "";

    if (in_array(3, $intersect)) {
        $result.="Pling";
    }

    if (in_array(5, $intersect)) {
        $result.="Plang";
    }

    if (in_array(7, $intersect)) {
        $result.="Plong";
    }

    return $result;
}

File: ruby/raindrops/raindrops.rb
class Raindrops

  def self.convert (number)

    result = ""

    (1..number).each do |divider|

      if (number % divider == 0) && divider == 3 then
        result << "Pling"
      elsif (number % divider == 0) && divider == 5 then
        result << "Plang"
      elsif (number % divider == 0) && divider == 7 then
        result << "Plong"
      end

    end

    if result.empty? then
      return number.to_s
    end

    return result
  end

end

module BookKeeping
  VERSION = 3
end